connectors/memory/memory_test.go

```diff
@@ -1173,6 +1173,234 @@ func TestCompoundPartSecondaryIndex(t *testing.T) {
 	assert.EqualValues(t, readVals1["CreatedAt"], createdAt1)
 }
 
+func TestReadRaceNonClustered(t *testing.T) {
```

Comment:
I am curious, will this test fail with `go test` in CI build, or we can only reproduce the race condition with `-race` flag on?

Reply:
Without the fix and without the `-race` flag on, this test will sometimes fail with just a `go test`, because one go-routine does the write before the read. If you then turn on -race, you'll see this problem pointed out by the race detector. With the problem now fixed though, this test will always pass, even if you don't run it with `-race`, because the actually problem is fixed.